lib/fog/kubevirt/compute/models/vms.rb

```diff
@@ -176,7 +176,7 @@ def add_vm_storage(vm_name, vm_volumes)
           normalized_vm_name = vm_name.gsub(/[._]+/,'-')
           volumes, disks = [], []
           vm_volumes.each_with_index do |v, idx|
-            volume_name = v.name || normalized_vm_name + "-disk-0" + idx.to_s
+            volume_name = v.name.nil? ? normalized_vm_name + "-disk-0" + idx.to_s : v.name.gsub(/[._]+/,'-')
```

Comment:
The expression `gsub(/[._]+/,'-')` is the same as in line 176, and we'd might this functionality for other restricted 
 resources names. How about extracting it to utils to a method, i.e. `normalized_name(name)`

Reply:
Utility sound ok. We have exactly the bug for it and we fixed it some time ago. With recent refactorings the bug is back :)